webapp/src/ts/services/cht-datasource.service.ts

```diff
@@ -144,4 +144,8 @@ export class CHTDatasourceService {
       }
     };
   }
+
+  getDataContext() {
+    return this.dataContext;
```

Comment:
issue: this `dataContext` will  be null until we actually init the service.  I think this function needs to be `async` and then before returning the data context we just do: `await this.isInitialized();`